spl/src/memo.rs

```diff
@@ -0,0 +1,29 @@
+use anchor_lang::solana_program::pubkey::Pubkey;
+use anchor_lang::Result;
+use anchor_lang::{context::CpiContext, Accounts};
+
+pub use spl_memo;
+pub use spl_memo::ID;
+
+pub fn build_memo<'info>(ctx: CpiContext<'_, '_, '_, 'info, BuildMemo>, memo: &[u8]) -> Result<()> {
+    let mut signer_pubkeys: Vec<&Pubkey> = vec![];
+    for i in 0..ctx.remaining_accounts.len() {
+        signer_pubkeys.push(ctx.remaining_accounts[i].key);
+    }
```

Comment:
I'm curious why you're using all of the `remaining_accounts` and there isn't a way for a user to provide signers from the context's list of specific accounts?

Fwiw, you could also do:

```rust
let signer_pubkeys = ctx.remaining_accounts.map(|account| account.key).collect::<Vec<_>>();
```

Reply:
@buffalojoec thanks for your feedback.
Memo instruction requires signers and they are all optional so I used remaining_accounts only.
You can also add memo w/o any signers.

Let me follow up with your map suggestion though. :pray: